src/rust/util/src/cell.rs

```diff
@@ -12,48 +12,44 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Same as `core::sync::SyncUnsafeCell`, which is still in nightly. Can remove once it is stable.
-pub struct SyncUnsafeCell<T: ?Sized> {
+// Needed because otherwise we have undefined references to _critical_section_1_0_acquire/release.
+// This is the official solution:
+// https://github.com/rust-embedded/critical-section/blob/bc0d1736c3b14e340c1b0c29042edb3e6bffe6b8/README.md#undefined-reference-errors
+#[cfg(not(any(feature = "testing", feature = "c-unit-testing")))]
+use cortex_m as _;
+
+// Inspired by as `core::sync::SyncUnsafeCell`, which is still in nightly. Critical section added so
```

Comment:
Is it really inspired by SyncUnsafeCell? Doesn't seem to have much in common.

Reply:
Started out as, but yeah I removed that now.